src/utils/tracked-controls.js

```diff
@@ -162,7 +162,7 @@ function findMatchingControllerWebXR (controllers, idPrefix, handedness, index,
   for (i = 0; i < controllers.length; i++) {
     controller = controllers[i];
     profiles = controller.profiles;
-    if (profiles.length === 0) { return; }
```

Comment:
Is this needed?

Reply:
The change from `return` to `continue`? Yes, this is the main fix to detect the controller, see the screenshot and my comments. The line in itself? Yes, because we use `profiles[0]` below.

Reply:
I'm worried about this code to break other controllers support. It needs to be tested properly in several browsers and headsets

Reply:
I think there might be a browser bug. I think the profiles array should be always populated for all controllers through WebXR. I would like to avoid workarounds browser bugs.

Reply:
This may be true. But in this case why did you have this check in the first place? You already had this code to workaround a browser bug I think. The check before didn't try other controllers, the new check now try the other controllers. Would you like me to remove this line and just replace 
```
controllerMatch = profiles[0].startsWith(idPrefix);
```
by
```
controllerMatch = profiles.length > 0 && profiles[0].startsWith(idPrefix);
````
? This will work too.

Even if I create an issue about the gaze controller having an empty profiles array on the Oculus browser tracker, we won't see any update of Oculus browser for Gear VR, so I think there is no point of creating it.

Reply:
Yes, that line was introduce by this [commit](https://github.com/aframevr/aframe/commit/9a7d7f41f9bb26addb7c405180fdbaceba265e4e). Profiles were not available when Oculus Browser shipped WebXR for the first time. We should first understand if it's a bug in the Oculus Browser implementation or expected behavior.

Reply:
The controllers detection code is complicated and workarounds for a device that has been discontinued might not be worth the risk

Reply:
I can understand this. So I'm ok to remove the line completely and add an additional check before calling startsWith on profiles[0] just in case profiles[0] is undefined. This is just js best practice and not a workaround. :)